src/power_grid_model/typing.py

```diff
@@ -5,8 +5,8 @@
 """
 Type hints for PGM. This includes all miscellaneous type hints not under dataset or dataset_definitions categories
 """
-from power_grid_model.core.dataset_definitions import ComponentType
+from power_grid_model.data_types import ComponentTypeLike
 
-_ComponentAttributeMappingDict = dict[ComponentType, set[str] | list[str] | None]
+_ComponentAttributeMappingDict = dict[ComponentTypeLike, set[str] | list[str] | None]
 
-ComponentAttributeMapping = set[ComponentType] | list[ComponentType] | None | _ComponentAttributeMappingDict
+ComponentAttributeMapping = set[ComponentTypeLike] | list[ComponentTypeLike] | None | _ComponentAttributeMappingDict
```

Comment:
actually, we can revert this to the original state. This is new behavior and we therefore can assume that people using it have migrated to the modern way of doing things